src/pubsub/ua_pubsub_ns0.c

```diff
@@ -1079,6 +1085,145 @@ publishedDataItemsTypeDestructor(UA_Server *server,
+    inputArgument.name = UA_STRING("BinFile");
+    inputArgument.dataType = UA_TYPES[UA_TYPES_BYTESTRING].typeId;
+    inputArgument.valueRank = UA_VALUERANK_SCALAR;
+
+    UA_MethodAttributes configAttr = UA_MethodAttributes_default;
+    configAttr.description = UA_LOCALIZEDTEXT("en-US","Load binary configuration file");
+    configAttr.displayName = UA_LOCALIZEDTEXT("en-US","LoadPubSubConfiguration");
+    configAttr.executable = true;
+    configAttr.userExecutable = true;
+    UA_StatusCode retVal = UA_Server_addMethodNode(server, UA_NODEID_STRING(1, "PubSub_replace_config_method"),
+                                                   UA_NODEID_NUMERIC(0, UA_NS0ID_PUBLISHSUBSCRIBE),
+                                                   UA_NODEID_NUMERIC(0, UA_NS0ID_HASORDEREDCOMPONENT),
+                                                   UA_QUALIFIEDNAME(1, "PubSub configuration"),
+                                                   configAttr, &UA_loadPubSubConfigMethodCallback,
+                                                   1, &inputArgument, 0, NULL, NULL, NULL);
+    return retVal;
+}
+#endif
+#endif /*UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS*/
+
+
+/* UA_deletePubSubConfigMethodCallback() */
+/**
+ * @brief   callback function that will be executed when the method "PubSub configurator (delete config)" is called.
+ */
+#ifdef UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS
+#ifdef UA_ENABLE_PUBSUB_CONFIG
+static UA_StatusCode
+UA_deletePubSubConfigMethodCallback(UA_Server *server,
+                                    const UA_NodeId *sessionId, void *sessionHandle,
+                                    const UA_NodeId *methodId, void *methodContext,
+                                    const UA_NodeId *objectId, void *objectContext,
+                                    size_t inputSize, const UA_Variant *input,
+                                    size_t outputSize, UA_Variant *output)
+{
+    UA_PubSubManager_delete(server, &(server->pubSubManager));
+    
+    return UA_STATUSCODE_GOOD;
+}
+#endif
+#endif /*UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS*/
+
+
+/* UA_addDeletePubSubConfigMethod() */
+/**
+ * @brief       Adds method node to server. This method is used to delete the current PubSub configuration.
+ * 
+ * @param       server      [bi]    UA_Server object that shall contain the method.
+ * 
+ * @return      UA_STATUSCODE_GOOD on success
+ */
+#ifdef UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS
+#ifdef UA_ENABLE_PUBSUB_CONFIG
+static UA_StatusCode 
+UA_addDeletePubSubConfigMethod(UA_Server *server)
+{
+    UA_MethodAttributes configAttr = UA_MethodAttributes_default;
+    configAttr.description = UA_LOCALIZEDTEXT("en-US","Delete current PubSub configuration");
+    configAttr.displayName = UA_LOCALIZEDTEXT("en-US","DeletePubSubConfiguration");
+    configAttr.executable = true;
+    configAttr.userExecutable = true;
+    UA_StatusCode retVal = UA_Server_addMethodNode(server, UA_NODEID_STRING(1, "PubSub_delete_config_method"),
```

Comment:
Static string node ids are uncommon. I think it would be better to autogenerate a numeric node ID.